R/getBestFeatureParameters.R

```diff
@@ -0,0 +1,431 @@
+#'                                                        hypothesis_completeness_cutoffs = c(0.5,1),
+#'                                                        n_subunits_cutoffs =c(2,3,4),
+#'                                                        monomer_distance_cutoffs = c(1,2),
+#'                                                        remove_decoys=FALSE)
+#' 
+#' ## Calculate the FDR statistic for every grid search result
+#' # Here we only performed grid search of different filter cutoffs, so only those are
+#' # specified as grid_search_params
+#' 
+#' gridStats <- estimateGridSearchDecoyFDR(complexFeaturesGridFiltered,
+#'                                         grid_search_params =c("min_feature_completeness",
+#'                                                               "min_hypothesis_completeness",
+#'                                                               "min_subunits",
+#'                                                               "min_peak_corr",
+#'                                                               "min_monomer_distance_factor"))
+#' ## Inspect the oputput                                                              
+#' gridStats
+
+estimateGridSearchDecoyFDR<- function(complex_features_list, 
+                                      grid_search_params =c("corr", "window",
+                                                            "rt_height", "smoothing_length",
+                                                            "min_feature_completeness",
+                                                            "min_hypothesis_completeness",
+                                                            "min_subunits",
+                                                            "min_peak_corr", 
+                                                            "min_monomer_distance_factor")){
+  
+  x=lapply(complex_features_list,estimateDecoyFDR, 
+           grid_search_params = grid_search_params,
+           verbose = FALSE)
+  
+  x_names = names(x[[1]])
+  y = as.data.table(t(setDT(x)))[,lapply(.SD,unlist)]
+  names(y)=x_names
+  y[]
+}
+
+#' Get the best performing ParameterStats
+#' @description Pick the parameter set in a grid search stats output that performs
+#' best in terms of identifications while staying within a specified FDR cutoff.
+#' @param grid_search_stats Table of grid search statistics.
+#' (obtained from \code{\link{estimateGridSearchDecoyFDR}}).
+#' @param FDR Numeric, maximum FDR that should be considered. Default = \code{0.1}.
+#' @return data.table one row with best stats across grid search.
+#' @export
+#' @examples
+#' 
+#' ## NOT RUN
+#' gridStats # see function \code{\link{estimateGridSearchDecoyFDR}} to see how to generate this object.
+#' 
+#' ## Pick best parameter set
+#' getBestParameterStats(gridStats)
+#' 
+
+getBestParameterStats <- function(grid_search_stats,
+                                  FDR = 0.1){
+  env<-environment()
```

Comment:
That's a good question. I just left it like it was, because I assumed you had a reason for doing it this way. But I also don't quite understand why.